org.lflang/src/org/lflang/generator/c/CGenerator.java

```diff
@@ -1962,6 +1962,14 @@ protected void setUpGeneralParameters() {
             System.out.println("To enable compilation for the Arduino platform, you must specify the fully-qualified board name (FQBN) in the target property. For example, platform: {name: arduino, board: arduino:avr:leonardo}. Entering \"no-compile\" mode and generating target code only.");
             targetConfig.noCompile = true;
         }
+
+        if (targetConfig.platformOptions.platform == Platform.ZEPHYR && targetConfig.platformOptions.userThreads >= 0) {
+            targetConfig.compileDefinitions.put(
+                "USER_THREADS",
```

Comment:
We should not use a plain string here as we have an enum to our disposal. In fact, this particular string doesn't match the enum, which is exactly why we shouldn't do this.